core/src/parking_lot.rs

```diff
@@ -80,27 +79,37 @@ struct FairTimeout {
     // Next time at which point be_fair should be set
     timeout: Instant,
 
-    // Random number generator for calculating the next timeout
-    rng: SmallRng,
+    // the PRNG state for calculating the next timeout
+    seed: u32,
 }
 
 impl FairTimeout {
     #[inline]
     fn new(timeout: Instant) -> FairTimeout {
-        FairTimeout { timeout, rng: SmallRng::from_entropy() }
+        FairTimeout { timeout, seed: &timeout as *const _ as usize as u32 }
```

Comment:
I'm not a big fan of using memory addresses as seeds since they could in theory be used to leak memory layout information to attackers and defeat address space randomization. Could we maybe use `timeout.nanos() + 1` as the initial seed instead?

Reply:
Sure. I just now realized other downsides with the address approach as well. Fixing :+1:

Reply:
Well, `Instant` has no `nanos` method. We would need to compute a `Duration` for that.

Reply:
We could just take the `seed` as argument and have `HashTable::new` just have an incrementing counter it sends. So first `FairTimeout` has seed `1` etc.

Reply:
Yes, an incrementing counter would work fine.